library/posture.c

```diff
@@ -21,22 +21,35 @@ limitations under the License.
 #define NANOS(s) ((s) * 1e9)
 #define MILLIS(s) ((s) * 1000)
 
+const int NO_TIMEOUTS = -1;
```

Comment:
I added a sentinel value in the controller for configurable timeouts. If it is the `timeout` property is set to -1 then:

1. time variability is supported
2. there are no timeouts on the posture checks if the timeout == -1
2)

Reply:
So if 

new controller + old SDK, the old SDK will over submit posture data, but there should be no issue with that

old controller + new SDK, the new SDK will detect the old controller not supporting timeouts and will submit as the old SDKs used to.